File: 백준/Silver/9095. 1， 2， 3 더하기/1， 2， 3 더하기.py
'''
dp[n] = dp[n-3] + dp[n-2] + dp[n-1]

n=1 -> 1
n=2 -> 2
n=3 -> 4
'''
n = int(input())

for i in range(n):
    m = int(input())
    
    dp=[0, 1, 2, 4]    # 0~3일 경우에 대해 초기화
    
    for i in range(4, m+1):
        dp.append(dp[i-1] + dp[i-2] + dp[i-3])    # 새로운 값을 배열에 삽입
    
    print(dp[m])

File: 백준/Silver/9095. 1， 2， 3 더하기/1， 2， 3 더하기.js
const readline = require("readline");

let t;
let nList = [];

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
});
rl.on("line", (line) => {
    if (!t) {
        t = Number(line);
    } else {
        nList.push(Number(line));

        if (nList.length === t) {
            rl.close();
        }
    }
}).on("close", () => {
    nList.map(n => {
        const dp = Array(n + 1).fill(0);
        dp[1] = 1;    // 1
        dp[2] = 2;    // 2
        dp[3] = 4;    // 3

        for (let i = 4; i <= n; i++) {
            dp[i] = dp[i - 1] + dp[i - 2] + dp[i - 3];
        }

        return console.log(dp[n]);
    });

    process.exit();
});